addons/sale/wizard/sale_order_discount.py

```diff
@@ -84,7 +84,7 @@ def _create_discount_lines(self):
                 self._prepare_discount_line_values(
                     product=discount_product,
                     amount=self.discount_amount,
-                    taxes=self.env['account.tax'],
+                    taxes=discount_product.taxes_id,
```

Comment:
I don't it would be this simple

Whether we apply tax or not amount for fix amount discount should be same
When I select 10$ amount discount and there is 10% tax then amount will be 11$ but it should be 10 after tax so you need to update amount before tax so your amount should be x where x=10 - (x%10)

Reply:
@Feyensv Could I have your opinion on this one? :slightly_smiling_face: